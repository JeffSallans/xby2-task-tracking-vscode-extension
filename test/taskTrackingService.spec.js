/* global suite, test */

//
// Note: This example test is leveraging the Mocha test framework.
// Please refer to their documentation on https://mochajs.org/ for help.
//

// The module 'assert' provides assertion methods from node
const fs = require('fs');
const moment = require('moment');

// You can import and use all API from the 'vscode' module
// as well as import your extension to test it
const taskTrackingService = require('../taskTrackingService');
const getProjects = require('./mock-requests/get-projects.json');
const getTasks = require('./mock-requests/get-tasks.json');
const getTaskDetails = fs.readFileSync('./test/mock-requests/get-task-details.html');
const getTasksMonthView = fs.readFileSync('./test/mock-requests/get-tasks-month-view.html');

jest.mock('httpntlm', () => ({
    get: jest.fn(() => ({})),
    post: jest.fn(() => ({})),
}));
const httpntlm = require('httpntlm');

describe("Task Tracking Service", () => {

    describe("getWeeklyTasks", () => {
        
        it('returns the correct data for the week of August 12 using mock data', () => {
            httpntlm.post.mockImplementationOnce((settings, callback) => {
                callback(null, {
                    statusCode: 200,
                    body: getTasksMonthView.toString(),
                });
            });
            return taskTrackingService.getWeeklyTasks('test-username', 'test-password', '08-02-2017')
            .then((weeklyTasks) => {
                expect(weeklyTasks).toMatchSnapshot();
            })
        });
    });

    describe("getMonthlyTasks", () => {
            
        it('returns the correct data for the week of August 12 using mock data', () => {
            httpntlm.post.mockImplementationOnce((settings, callback) => {
                callback(null, {
                    statusCode: 200,
                    body: getTasksMonthView.toString(),
                });
            });
            return taskTrackingService.getMonthlyTasks('test-username', 'test-password', '08-02-2017')
            .then((monthlyTasks) => {
                expect(monthlyTasks).toMatchSnapshot();
            })
        });
    });

    describe("getClients", () => {
        
        it('returns the correct data for the week of August 12 using mock data', () => {
            httpntlm.get.mockImplementationOnce((settings, callback) => {
                callback(null, {
                    statusCode: 200,
                    body: getTaskDetails.toString(),
                });
            });
            return taskTrackingService.getClients('test-username', 'test-password', '08-02-2017')
            .then((clients) => {
                expect(clients).toMatchSnapshot();
            })
        });
    });
});
