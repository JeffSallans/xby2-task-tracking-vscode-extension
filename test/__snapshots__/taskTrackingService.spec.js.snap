// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Tracking Service getClients returns the correct data for the week of August 12 using mock data 1`] = `
Array [
  Object {
    "Id": 27,
    "Name": "X by 2",
    "isDefault": false,
  },
  Object {
    "Id": 48,
    "Name": "Auto-Owners Insurance",
    "isDefault": true,
  },
]
`;

exports[`Task Tracking Service getMonthlyTasks returns the correct data for the week of August 12 using mock data 1`] = `
Array [
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-01T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-02T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-03T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-04T04:00:00.000Z",
    "description": "",
    "hours": 4,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
]
`;

exports[`Task Tracking Service getWeeklyTasks returns the correct data for the week of August 12 using mock data 1`] = `
Array [
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-07-31T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-01T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-02T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-03T04:00:00.000Z",
    "description": "",
    "hours": 8,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
  Object {
    "activityId": null,
    "clientId": null,
    "clientName": "Auto-Owners Insurance",
    "date": "2017-08-04T04:00:00.000Z",
    "description": "",
    "hours": 4,
    "isBillable": true,
    "minutes": 0,
    "projectId": null,
    "projectName": "",
    "taskId": null,
    "taskName": "",
  },
]
`;
